config/schema.yml

```diff
@@ -287,9 +287,9 @@ propel:
 
   relation:
     id: { type: integer, required: true, primaryKey: true, foreignTable: object, foreignReference: id, onDelete: cascade, inheritanceKey: true }
-    subject_id: { type: integer, required: true, foreignTable: object, foreignReference: id }
-    object_id: { type: integer, required: true, foreignTable: object, foreignReference: id }
-    type_id: { type: integer, foreignTable: term, foreignReference: id }
+    subject_id: { type: integer, required: true, foreignTable: object, foreignReference: id, onDelete: cascade }
+    object_id: { type: integer, required: true, foreignTable: object, foreignReference: id, onDelete: cascade }
+    type_id: { type: integer, foreignTable: term, foreignReference: id, onDelete: cascade }
```

Comment:
This one worries me a bit. In principle I think it's true that a relation without at `type_id` doesn't mean much, my concern is that a user deletes a term without realizing it will also delete all related relation rows.  It's a tough choice but I lean slightly towards "onDelete: setnull" for type_id to avoid unintended relation deletes.

Reply:
I kind of agree with @djjuhasz but I'd like to hear your opinion in this one, @fiver-watson.

Reply:
Not sure I'm totally understanding this... What would be the advantage of leaving a relation row that points to a NULL, @djjuhasz  - can you think of real-world examples where someone would want to delete a term but still have references to the missing term in the related entities? And, would the user have any option for deleting that relation once the term is gone? Thanks1

Reply:
Current functionality appears to be 'ON DELETE RESTRICT' which will prevent deletion of terms for which relations exist.

Dan raises a good point - if relation.type_id is NULL, then the relation will be ignored in any queries that filter on types of relations.

Reply:
@fiver-watson I can't think of any use case where having a relation row with no type_id is desirable.  

My concern is that somebody deleting a term (especially via direct MySQL access rather than the UI) that is used for a relation.type_id, without realizing that it's going to delete every relationship that uses that term.  E.g. a user naively deletes the "controls" term, and wipes out all "controls" relationships.  At least if the relation.type_id is set to null, the user can recover by adding a new type_id to the row. After deleting the term, the user could delete all relation rows where relation.type_id = NULL if that's the objective.

It's a tough call though, allowing relation.type_id to be null does have the potential to hide those relation rows from the UI and in effect make them hidden data.

Reply:
I've talked about this a bit more with @jraddaoui and we are both still kind of leaning towards ON DELETE CASCADE. In the UI, we at least give users a warning page before a deletion (though with terms we don't give a count of affected relations). In the DB, we **always** recommend making a backup before directly manipulating the data using SQL - i'm not sure that leaving empty rows to account for potentially undoing user error is a good tradeoff in terms of overall database consistency. 

We could potentially consider rewording the warning screen that appears when a term is deleted, to add a clearer warning? Even without a count, that might be useful. 

Ultimately, i think this is a decision that I trust you all to come to consensus around, as it feels more about project maintenance than end user experience and usage. I think your argument is an interesting one David, but what would it mean if we applied it consistently throughout AtoM?

Reply:
Okay, I'm fine with ON DELETE CASCADE - I agree that it's better for database consistency and I'm probably being over-cautious in worrying about unanticipated deletes.

Reply:
On the topic of database consistency, I wonder if it's possible to do an ON DELETE CASCADE of the `object` row when a related object is deleted?  I think it would require putting a foreign key constraint on e.g. `information_object.id` to `object.id` which may cause a circular dependency, but it would sure be nice to avoid leaving behind useless `object` table rows.

Reply:
Hi @djjuhasz, I'm not sure how that would work in the ORM but I think @fiver-watson added some notes about this in the SQL work he did. I'd prefer not to go that way and ask to do the deletes from the object row, otherwise we may want to do the same in other inheritances (QubitRepository and others from QubitActor for example) and it could get messy.

Reply:
And thanks all for the discussion.